### PR TITLE
feat(segmentation): Enhanced segmentation panel design for TMTV 

### PR DESCRIPTION
**Pull request by IbrahimCSAE**
### Context

Addresses issue #3790 

### Changes & Results

Pending

### Screenshots

image

image

image

image

image

### Notes

Most work is complete, remaining piece is to switch out the TMTV panel

### CODE REVIEW

File: platform/ui/src/assets/styles/styles.css [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* CUSTOM OHIF SCROLLBAR */
 .ohif-scrollbar {
   scrollbar-color: #173239 transparent;
-  scrollbar-gutter: stable;
+  overflow-y: auto;
```
Comment (sedghi) [inline platform/ui/src/assets/styles/styles.css:4]: i'm not sure why you are removing the gutter, it is needed to overlay the thumnail on top of the div, no?

File: extensions/cornerstone-dicom-seg/src/panels/PanelSegmentation.tsx [hunk 2]
```diff
@@ -224,54 +224,105 @@ export default function PanelSegmentation({
...
-            _setSegmentationConfiguration(selectedSegmentationId, 'outlineWidthActive', value)
-          }
-          setFillAlpha={value =>
-            _setSegmentationConfiguration(selectedSegmentationId, 'fillAlpha', value)
-          }
-          setFillAlphaInactive={value =>
-            _setSegmentationConfiguration(selectedSegmentationId, 'fillAlphaInactive', value)
-          }
-        />
+        {configuration.segmentationPanelMode === 'dropdown' ? (
```
Comment (sedghi) [inline extensions/cornerstone-dicom-seg/src/panels/PanelSegmentation.tsx:227]: this looks redundant, can you not do it here, and pass the mode inside ui and do it there?

File: platform/ui/src/components/SegmentationGroupTable/SegmentationDropDownRow.tsx [hunk 3]
```diff
@@ -122,8 +122,24 @@ function SegmentationDropDownRow({
...
+            <div className="flex w-[140px] flex-col">
+              <div className="text-[13px] text-white">Series:</div>
+              <div className="text-aqua-pale text-[13px]">{activeSegmentation.description}</div>
+            </div>
+          }
+        >
+          <Icon
+            name="info-action"
+            style={{
+              color: '#348CFD',
```
Comment (sedghi) [inline platform/ui/src/components/SegmentationGroupTable/SegmentationDropDownRow.tsx:137]: please don't hard code the color, if we need a new color add it to tailwind config

**Review by sedghi — COMMENTED** [review]
see my comments please, great work

File: extensions/cornerstone-dicom-seg/src/panels/PanelSegmentation.tsx (hunk 2 above)
Comment (IbrahimCSAE) [reply extensions/cornerstone-dicom-seg/src/panels/PanelSegmentation.tsx:227]: I did something different, let me know if that's fine

**Review by sedghi — COMMENTED** [review]
can we make the scrollbar outside the header 

CleanShot 2024-03-27 at 12 22 50@2x

----



number box is too big
CleanShot 2024-03-27 at 12 23 05@2x


---

needs to be half 

CleanShot 2024-03-27 at 12 23 28@2x


---

can you add the new segmetnation to the tmtv 

CleanShot 2024-03-27 at 12 24 05@2x

**Comment by IbrahimCSAE** [comment]
Fixes:

1. Scrollbar outside header for dropdown and expanded (I needed to set max height 500px]

image

image

2.number box sizing fixed

image

3. Reload data patient info size is corrected and smaller

image

4. Added new segmentation panel to tmtv (some features do not work as I'm not sure how it should function in tmtv)

image

5. Fixed visibility toggle bug, now it works as expected

6. fixed I icon location when expanded/unexpanded

image

7. scroll bars start after header now

8.Fixed add segment padding

image

9. working on it

10. centered settings icon

image

11. Fixed add segmentation padding

image


@sedghi please take a look, we just need to add the tools and fix up panel in tmtv.

File: extensions/default/src/Toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -33,16 +31,7 @@ export function Toolbar({ servicesManager }) {
           />
         );
 
-        return disabled ? (
```
Comment (IbrahimCSAE) [inline extensions/default/src/Toolbar/Toolbar.tsx:36]: duplicate tooltip bug

File: platform/ui/src/components/Tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -67,37 +154,7 @@ const Tooltip = ({
...
-            viewBox="0 0 24 24"
-          >
-            <path
-              fill="currentColor"
-              d="M24 22l-12-20l-12 20"
-            />
-          </svg>
-        </div>
-      </div>
+      {tooltipContainer && ReactDOM.createPortal(tooltipContent, tooltipContainer)}
```
Comment (IbrahimCSAE) [inline platform/ui/src/components/Tooltip/Tooltip.tsx:157]: react portals

File: modes/segmentation/src/segmentationButtons.ts [hunk 6]
```diff
@@ -95,7 +95,7 @@ const toolbarButtons: Button[] = [
         {
           id: 'Threshold',
           icon: 'icon-tool-threshold',
-          label: 'Eraser',
+          label: 'Threshold Tool',
```
Comment (IbrahimCSAE) [inline modes/segmentation/src/segmentationButtons.ts:98]: fix tool name

**Comment by IbrahimCSAE** [comment]
image

react portals in action, looks good

File: platform/ui/src/components/Tooltip/Tooltip.tsx [hunk 7]
```diff
@@ -57,8 +62,90 @@ const Tooltip = ({
 
   const isOpen = (isSticky || isActive) && !isDisabled;
 
+  useEffect(() => {
+    if (isOpen && parentRef.current && tooltipRef.current) {
```
Comment (sedghi) [inline platform/ui/src/components/Tooltip/Tooltip.tsx:66]: Nice! 

Can you please add a comment on what is happening for future devs, newX, newY, parent ...

Comment (sedghi) [reply platform/ui/src/components/Tooltip/Tooltip.tsx:66]: if (!isOpen || !parentRef.current || !tooltipRef.current){
  return
}

--> less indentation --> more clear

**Review by sedghi — COMMENTED** [review]
Getting there, 

Patient info has wrong icon 

CleanShot 2024-04-02 at 09 10 35@2x


---------
The dropdown for the modal seems like getting in the div (maybe we need portal here as well?)

https://github.com/OHIF/Viewers/assets/7490180/b47619bb-2440-4414-b275-a29b9177db1f


--------

Empty Segmentation should have one segment at least, so we should add segment 1 programmatically

https://github.com/OHIF/Viewers/assets/7490180/3287ae87-4432-4ea8-aadc-77435960ccbf


----

suv peak display text does not appear for some reason below the segment index


https://github.com/OHIF/Viewers/assets/7490180/197f6fc4-50d4-4fd5-86c8-5bb94a7a11eb

----

seems like active segment cannot be ch

https://github.com/OHIF/Viewers/assets/7490180/58535523-b804-4331-9f79-41be172d9c77

anged

**Pull request merged**